pkg/templates/git.go

```diff
@@ -456,3 +461,37 @@ func getValidVersions(
 
 	return validVersions, versionSchema, nil
 }
+
+// SyncGitSSLSkipVerify syncs ssl skip verify setting.
+// It will set environment variable GIT_SSL_NO_VERIFY.
+func SyncGitSSLSkipVerify(ctx context.Context, m settingbus.BusMessage) error {
+	for i := 0; i < len(m.Refers); i++ {
+		if m.Refers[i] == nil || m.Refers[i].Name == "" {
+			continue
+		}
+
+		if m.Refers[i].Name != settings.SSLSkipVerify.Name() {
+			continue
+		}
+
+		err := SetGitCAEnvVar(ctx, m.TransactionalModelClient)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+// SetGitCAEnvVar sets environment variable GIT_SSL_NO_VERIFY.
+func SetGitCAEnvVar(ctx context.Context, mc model.ClientSet) error {
+	// Set environment variable GIT_SSL_NO_VERIFY.
+	err := os.Setenv(GitSSLSkipVerifyKey, settings.SSLSkipVerify.ShouldValue(ctx, mc))
+	if err != nil {
+		return err
+	}
+
+	log.WithName("git").Infof("set environment variable %s", GitSSLSkipVerifyKey)
+
+	return nil
+}
```

Comment:
the getter client has an option for insecure verification skipping, but the GitGetter implementation does not respect this option. I think we can fork the [go-getter](https://github.com/hashicorp/go-getter) from hashicorp,  implement that and push to upstream.

https://github.com/hashicorp/go-getter/blob/0298a221674f629339295fa8a1e6a938e28506e0/get_git.go#L206-L207

Reply:
updated